pallets/ethereum-xcm/src/lib.rs

```diff
@@ -228,6 +230,39 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Xcm Transact an Ethereum transaction, but allow to force the caller and create address.
+		/// This call should be restricted (callable only by the runtime or governance).
+		/// Weight: Gas limit plus the db reads involving the suspension and proxy checks
+		#[pallet::weight({
+			let without_base_extrinsic_weight = false;
+			<T as pallet_evm::Config>::GasWeightMapping::gas_to_weight({
+				match xcm_transaction {
+					EthereumXcmTransaction::V1(v1_tx) =>  v1_tx.gas_limit.unique_saturated_into(),
+					EthereumXcmTransaction::V2(v2_tx) =>  v2_tx.gas_limit.unique_saturated_into()
+				}
+			}, without_base_extrinsic_weight).saturating_add(T::DbWeight::get().reads(2))
```

Comment:
I think we don't need to account for the proxy checks here
```suggestion
			}, without_base_extrinsic_weight).saturating_add(T::DbWeight::get().reads(1))
```